c10/xpu/XPUStream.cpp

```diff
@@ -0,0 +1,266 @@
+          c10::Device(DeviceType::XPU, device_index),
+          stream_id));
+}
+
+} // anonymous namespace
+
+int XPUStream::priority() const {
+  StreamId stream_id = stream_.id();
+  StreamIdType st = streamIdType(stream_id);
+  // StreamIdType and priority number are inversely related.
+  return -static_cast<int>(st);
+}
+
+// See Note [StreamId assignment]
+sycl::queue& XPUStream::queue() const {
+  DeviceIndex device_index = stream_.device_index();
+  StreamId stream_id = stream_.id();
+  StreamIdType st = streamIdType(stream_id);
+  StreamIdIndex si = streamIdIndex(stream_id);
+  switch (st) {
+    case StreamIdType::NORMAL:
+    case StreamIdType::HIGH:
+      return *streams[device_index][static_cast<uint8_t>(st)][si];
+    default:
+      TORCH_CHECK(
+          false,
+          "Unrecognized stream ",
+          stream_,
+          " (I didn't recognize the stream type, ",
+          st,
+          ").",
+          " Did you manufacture the StreamId yourself?  Don't do that;");
+  }
+}
+
+// Returns a stream from the requested pool
+// Note: when called the first time on a device, this will create the stream
+// pools for that device.
+XPUStream getStreamFromPool(const int priority, DeviceIndex device) {
+  initXPUStreamsOnce();
+  if (device == -1) {
+    device = c10::xpu::current_device();
+  }
+  check_device(device);
+  TORCH_CHECK(
+      priority <= 0,
+      "Expected XPU stream priority to be less than or equal to 0, got ",
+      priority);
+  // Initializes the stream pools (once)
+  initDeviceStreamOnce(device);
+  auto priority_idx =
+      std::min(-priority, max_compile_time_stream_priorities - 1);
+  const auto idx = get_idx(priority_counters[device][priority_idx]);
+  auto id_type = static_cast<StreamIdType>(priority_idx);
+  return XPUStreamForId(device, makeStreamId(id_type, idx));
+}
+
+// Note: when called the first time on a device, this will create the stream
+// pools for that device.
+XPUStream getCurrentXPUStream(DeviceIndex device) {
+  initXPUStreamsOnce();
+  if (device == -1) {
+    device = c10::xpu::current_device();
+  }
+  check_device(device);
+  // Initializes the stream pool (once)
+  initDeviceStreamOnce(device);
+  return XPUStreamForId(device, current_streams[device]);
+}
+
+XPUStream getStreamFromPool(const bool isHighPriority, DeviceIndex device) {
+  initXPUStreamsOnce();
+  // If isHighPriority is true, return the stream with the highest priority.
+  int priority = isHighPriority ? -max_compile_time_stream_priorities + 1 : 0;
+  return getStreamFromPool(priority, device);
+}
+
+void setCurrentXPUStream(XPUStream stream) {
+  initXPUStreamsOnce();
+  current_streams[stream.device_index()] = stream.id();
+}
+
+std::ostream& operator<<(std::ostream& stream, const XPUStream& s) {
+  return stream << s.unwrap();
+}
+
+// Note: when called the first time on a device, this will create the stream
+// pools for that device.
+void streams_synchronize_on_device(DeviceIndex device) {
```

Comment:
Pls. keep same coding stype. Should be syncStreamsOnDevice ?

Reply:
Can we give more comments for this API?
Including, its usage, and our motivation?

Reply:
rename to `syncStreamsOnDevice`. Some comments are added in `XPUStreams.h` file.